src/reconcile/rec_visibility.c

```diff
@@ -568,8 +598,12 @@ __rec_upd_select(WT_SESSION_IMPL *session, WTI_RECONCILE *r, WT_UPDATE *first_up
         if ((txnid = upd->txnid) == WT_TXN_ABORTED)
             continue;
 
-        /* Give up if the update is from this transaction and on the metadata file.*/
-        if (WT_IS_METADATA(session->dhandle) && txnid == session_txnid)
+        /*
+         * Give up if the update is from this transaction and on the metadata file or disaggregated
+         * shared metadata file.
+         */
+        if ((WT_IS_METADATA(session->dhandle) || WT_IS_DISAGG_META(session->dhandle)) &&
+          txnid != WT_TXN_NONE && txnid == session_txnid)
```

Comment:
Can this txnid != WT_TXN_NONE the difference?

Reply:
I'll try it!